src/zoo/AnimalFactory.java

```diff
@@ -1,23 +1,55 @@
 package zoo;
 
-class AnimalFactory
+import animals.*;
+import exceptions.NoSuchAnimalException;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.util.logging.FileHandler;
+
+
+public class AnimalFactory
 {
-    /**
-     * Creates and returns an instance of a specific animal.
-     *
-     * @param animalName The animal to generate.
-     * @return the generated animal.
-     */
-    Animal generateAnimal(String animalName)
+
+    HashMap<String, Animal> animalMap = new HashMap<>();
+
+    Animal generateAnimal(String animalName) throws NoSuchAnimalException
     {
-        Animal newAnimal = switch (animalName)
-        {
+        Animal newAnimal = switch (animalName) {
             case "Duck" -> new Duck();
             case "Cat" -> new Cat();
             case "Dog" -> new Dog();
-            default -> null;
+            default -> throw new NoSuchAnimalException(animalName);
         };
 
         return newAnimal;
     }
-}
+
+    public ArrayList<Animal> parseAnimals(ArrayList<String> animalNames)
```

Comment:
read about encapsulation
generate animal should call parseAnimals

Reply:
the default if public/private isnt written
+should be private + under public functions

Reply:
read about stream/map